tests/test.cpp

```diff
@@ -50,7 +50,12 @@ uint32_t lcg_rand() {
 }
 void lcg_reset() { lcg_seed = 48271; }
 
-std::string test_data_path = "tests/";
+std::string test_data_path =
+#ifdef BAZEL_TEST_DATA_PATH
+    "../com_github_google_flatbuffers/tests/";
```

Comment:
Is this always how Bazel does paths? Would it be nicer to do `-DBAZEL_TEST_DATA_PATH=../com_github_google_flatbuffers/tests/` so this string is not hardcoded in the the source?

Reply:
There are a bunch of ways of doing this, but it boils down to it's not going to change going forwards unless pretty braking changes are made.

```
austin[108343] aschuh-debian-x270 (flatbuffer_cc) ~/local/flatbuffers/bazel-bin/flatbuffers_test.runfiles
$ find .
.
./com_github_google_flatbuffers
./com_github_google_flatbuffers/flatbuffers_test
./com_github_google_flatbuffers/tests
./com_github_google_flatbuffers/tests/include_test
./com_github_google_flatbuffers/tests/include_test/include_test1.fbs
./com_github_google_flatbuffers/tests/include_test/sub
./com_github_google_flatbuffers/tests/include_test/sub/include_test2.fbs
./com_github_google_flatbuffers/tests/monster_test.bfbs
./com_github_google_flatbuffers/tests/monster_test.fbs
./com_github_google_flatbuffers/tests/monsterdata_test.golden
./com_github_google_flatbuffers/tests/prototest
./com_github_google_flatbuffers/tests/prototest/imported.proto
./com_github_google_flatbuffers/tests/prototest/test.golden
./com_github_google_flatbuffers/tests/prototest/test.proto
./com_github_google_flatbuffers/tests/prototest/test_union.golden
./com_github_google_flatbuffers/tests/union_vector
./com_github_google_flatbuffers/tests/union_vector/union_vector.fbs
./com_github_google_flatbuffers/tests/unicode_test.json
./MANIFEST
```

The tests are run from the ```com_github_google_flatbuffers``` folder.  This constant comes from ```WORKSPACE``` name, which shouldn't be changing going forwards.

If you build the tests as part of another repo, all the test data will be in ```com_github_google_flatbuffers``` anyways.  The test will be run from the folder of the main repo (```frc971_org``` for us).  So, in both cases, ```../com_github_google_flatbuffers``` gets you back to the right folder.

I can pull it out into the BUILD file if you think that would be better.

Reply:
That was just an idea. If this is really always the case then I guess it is fine as is.